File: apps/backend/read-excel.py
#!/usr/bin/env python3
import sys
import json

try:
    import openpyxl
    import pandas as pd
except ImportError:
    print("Installing required packages...")
    import subprocess
    subprocess.check_call([sys.executable, "-m", "pip", "install", "--user", "openpyxl", "pandas"])
    import openpyxl
    import pandas as pd

# Read the Excel file
excel_file = '/Users/abdulsar/Downloads/Grants Master Sheet.xlsx'
xl = pd.ExcelFile(excel_file)

print(f"Sheet names: {xl.sheet_names}\n")

for sheet_name in xl.sheet_names:
    df = pd.read_excel(excel_file, sheet_name=sheet_name)
    print(f"\n{'='*60}")
    print(f"Sheet: {sheet_name}")
    print(f"{'='*60}")
    print(f"Rows: {len(df)}, Columns: {len(df.columns)}")
    print(f"\nColumn names:")
    for i, col in enumerate(df.columns, 1):
        print(f"  {i}. {col}")

    print(f"\nFirst 3 rows preview:")
    print(df.head(3).to_string())
    print("\n")

File: apps/backend/read-excel.js
const XLSX = require('xlsx');

// Read the Excel file
const workbook = XLSX.readFile('/Users/abdulsar/Downloads/Grants Master Sheet.xlsx');

console.log('Sheet names:', workbook.SheetNames);
console.log('\n' + '='.repeat(60));

workbook.SheetNames.forEach((sheetName) => {
  console.log(`\nSheet: ${sheetName}`);
  console.log('='.repeat(60));

  const worksheet = workbook.Sheets[sheetName];
  const data = XLSX.utils.sheet_to_json(worksheet, { header: 1 });

  console.log(`Rows: ${data.length}`);

  if (data.length > 0) {
    console.log('\nColumn headers:');
    const headers = data[0];
    headers.forEach((header, index) => {
      console.log(`  ${index + 1}. ${header}`);
    });

    console.log('\nFirst 3 data rows:');
    for (let i = 1; i <= Math.min(3, data.length - 1); i++) {
      console.log(`\nRow ${i}:`);
      const row = data[i];
      headers.forEach((header, index) => {
        if (row[index] !== undefined && row[index] !== null && row[index] !== '') {
          console.log(`  ${header}: ${row[index]}`);
        }
      });
    }
  }
  console.log('\n');
});
